VtGateV3 Lookup testcases

ajeetj opened the pull request:
Migrated VtGateV3 lookup testcases from Python to Go. 
Summary of tests:

- Test non-unique vindexes
- Test select by id
- Test select by lookup
- Test IN clause using non-unique vindex
- Test unowned functional vindex
- Test updates to secondary vindexes
- Test updating both vindexes
- Test updating only one vindex
- Test owned lookup unique index
- Test unowned lookup unique index


Detail about each case is present in below sheet:
https://docs.google.com/spreadsheets/d/1fMv0ehVMeQNVvjl3HAzZ7LuHzRaIZrjwQrENF0O3iE8/edit#gid=0

ajeetj committed:
added secondary lookup test cases

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj committed:
added secondary lookup test cases

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj committed:
added unowned and owned lookup tests

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj committed:
Remove commented code

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj commented:
@deepthi I still need to understand a few cases which I have commented out [We can discuss in our next syncup]. Besides that this PR is good to go.

deepthi reviewed (commented):
Nice work! Looks good on a first pass.
I will look at the details in the spreadsheet and do a more thorough review once I get access to it.

ajeetj committed:
updated scatter update test

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj commented:
Thanks @deepthi 
I also updated one test and converted it into error handling. 

There is 1 question/verification pending from my side where a duplicate insert into a unique column should fail, but it is not. I have added a TODO and will verify again when we have a new local cluster ready [Arindam is currently working on that]. 

This PR is good to merge now.

deepthi commented on go/vt/vtgate/endtoend/lookup_test.go:
```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
+
+	// insert multiple values
+	exec(t, conn, "begin")
+	exec(t, conn, "insert into t3(user_id, lastname, address) values(1,'snow','castle_black'), (2,'stark','winterfell')")
```
haha

deepthi commented on go/vt/vtgate/endtoend/lookup_test.go:
```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
+	exec(t, conn, "insert into t3(user_id, lastname, address) values(1,'snow','castle_black'), (2,'stark','winterfell')")
+	exec(t, conn, "commit")
+	qr := exec(t, conn, "select * from t3")
+	if got, want := fmt.Sprintf("%v", qr.Rows), "[[INT64(1) VARCHAR(\"snow\") VARCHAR(\"castle_black\")] [INT64(2) VARCHAR(\"stark\") VARCHAR(\"winterfell\")]]"; got != want {
```
Is the order of rows expected to be stable?

ajeetj replied:
Added order by clause

deepthi commented on go/vt/vtgate/endtoend/lookup_test.go:
```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//update both videxes
```
typo: vindexes

deepthi commented on go/vt/vtgate/endtoend/lookup_test.go:
```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
+	}
+}
+
+/*
```
Use // style doc comments instead of multi-line comments.

deepthi commented on go/vt/vtgate/endtoend/lookup_test.go:
```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
+	}
+}
+
+/*
```
Same comment as above.

deepthi reviewed (commented):
Looks good except for a few nit-picks.

deepthi reviewed (approved):
Can merge after addressing feedback.

ajeetj committed:
review comments addressed

Signed-off-by: Ajeet jain <ajeet@planetscale.com>

ajeetj commented:
> Can merge after addressing feedback.

Addressed review comments, merging now.
